src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -20,126 +20,119 @@ public static partial class Ascii
         /// <param name="right">The buffer to compare with <paramref name="left" />.</param>
         /// <returns><see langword="true" /> if the corresponding elements in <paramref name="left" /> and <paramref name="right" /> were equal and ASCII. <see langword="false" /> otherwise.</returns>
         /// <remarks>If both buffers contain equal, but non-ASCII characters, the method returns <see langword="false" />.</remarks>
+        public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
+            => left.Length == right.Length
+            && Equals<byte, byte, PlainLoader<byte>>(ref MemoryMarshal.GetReference(left), ref MemoryMarshal.GetReference(right), (uint)right.Length);
+
+        /// <inheritdoc cref="Equals(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
         public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length
+            && Equals<byte, ushort, WideningLoader>(ref MemoryMarshal.GetReference(left), ref Unsafe.As<char, ushort>(ref MemoryMarshal.GetReference(right)), (uint)right.Length);
+
+        /// <inheritdoc cref="Equals(ReadOnlySpan{byte}, ReadOnlySpan{char})"/>
+        public static bool Equals(ReadOnlySpan<char> left, ReadOnlySpan<byte> right)
+            => Equals(right, left);
+
+        /// <inheritdoc cref="Equals(ReadOnlySpan{byte}, ReadOnlySpan{char})"/>
+        public static bool Equals(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length
+            && Equals<ushort, ushort, PlainLoader<ushort>>(ref Unsafe.As<char, ushort>(ref MemoryMarshal.GetReference(left)), ref Unsafe.As<char, ushort>(ref MemoryMarshal.GetReference(right)), (uint)right.Length);
+
+        private static bool Equals<TLeft, TRight, TLoader>(ref TLeft left, ref TRight right, nuint length)
+            where TLeft : unmanaged, INumberBase<TLeft>
+            where TRight : unmanaged, INumberBase<TRight>
+            where TLoader : ILoader<TLeft, TRight>
```

Comment:
should we restrict it to `struct`?

```suggestion
            where TLoader : struct, ILoader<TLeft, TRight>
```

Reply:
Yeah, I think it makes sense to prevent perf-bugs very early due the struct-constraint.